coordinator/pipeline.go

```diff
@@ -523,8 +523,13 @@ func (p *Pipeline) forgeBatch(batchNum common.BatchNum) (batchInfo *BatchInfo,
 		if err != nil {
 			return nil, nil, tracerr.Wrap(err)
 		}
+		_l1UserFrozenTxs, err := p.historyDB.GetUnforgedL1UserTxs(p.state.lastForgeL1TxsNum + 2) //nolint:gomnd
```

Comment:
As discussed AFK, you should get all the queues `>= p.state.lastForgeL1TxsNum + 2` instead of just that one.
Also, you should pass the pendingL1UserTxs in the `GetL2TxSelection`

Reply:
Updated, using also a new method of historydb to get the future l1usertxs that will be forged in the future batch (except the next one)